src/action/filesystems.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! A handler and associated types for the filesystems action.
+//!
+//! The filesystems action lists all mounted filesystems on the client,
+//! collecting device name, mount point, filesystem type and its options.
+//! Current implementation works only in Linux systems.
+
+use rrg_proto::{Filesystem, KeyValue, AttributedDict, DataBlob};
+use crate::session::{self, Session};
+
+/// A response type for the filesystems action.
+pub struct Response {
+    /// Information about filesystem.
+    mount_info: proc_mounts::MountInfo,
+}
+
+/// Handles requests for the filesystems action.
+/// Initially searches in `/proc/mounts`. If it's missed, falls back to
+/// `/etc/mtab`.
+pub fn handle<S: Session>(session: &mut S, _: ()) -> session::Result<()> {
+    use proc_mounts::MountIter;
+
+    let mount_iter = match MountIter::new() {
+        Ok(mount_iter) => mount_iter,
+        Err(_) => {
+            // /etc/mtab must exist for sure.
+            MountIter::new_from_file("/etc/mtab").expect("/etc/mtab not found")
+        },
+    };
+
+    for mount_info in mount_iter {
+        session.reply(Response {
+            mount_info: mount_info.unwrap(),
```

Comment:
Now I see, that the underlying type of `MountIter` iterator is `io::Result<MountInfo>` ([source code](https://docs.rs/proc-mounts/0.2.4/src/proc_mounts/mounts/iter.rs.html#58)). That's why the answer is no, it can crash for some reason. I definitely have to improve error handling.